nemo/collections/tts/models/fastspeech2.py

```diff
@@ -0,0 +1,244 @@
+            tb_logger = self.logger.experiment
+            if isinstance(self.logger, LoggerCollection):
+                for logger in self.logger:
+                    if isinstance(logger, TensorBoardLogger):
+                        tb_logger = logger.experiment
+                        break
+            spec_target, spec_predict = training_step_outputs[0]["outputs"]
+            tb_logger.add_image(
+                "train_mel_target",
+                plot_spectrogram_to_numpy(spec_target[0].data.cpu().numpy()),
+                self.global_step,
+                dataformats="HWC",
+            )
+            spec_predict = spec_predict[0].data.cpu().numpy()
+            tb_logger.add_image(
+                "train_mel_predicted", plot_spectrogram_to_numpy(spec_predict.T), self.global_step, dataformats="HWC",
+            )
+            self.log_train_images = False
+
+    def validation_step(self, batch, batch_idx):
+        f, fl, t, tl, _, _, _ = batch
+        spec, spec_len = self.audio_to_melspec_preprocessor(f, fl)
+        mel, log_dur_preds, _, _, _ = self(spec_len=spec_len, text=t, text_length=tl)
+        loss = self.loss(spec_pred=mel, spec_target=spec, spec_target_len=spec_len, pad_value=-11.52)
+        return {
+            "val_loss": loss,
+            "mel_target": spec,
+            "mel_pred": mel,
+        }
+
+    def validation_epoch_end(self, outputs):
+        if self.logger is not None and self.logger.experiment is not None:
+            tb_logger = self.logger.experiment
+            if isinstance(self.logger, LoggerCollection):
+                for logger in self.logger:
+                    if isinstance(logger, TensorBoardLogger):
+                        tb_logger = logger.experiment
+                        break
+            _, spec_target, spec_predict = outputs[0].values()
+            tb_logger.add_image(
+                "val_mel_target",
+                plot_spectrogram_to_numpy(spec_target[0].data.cpu().numpy()),
+                self.global_step,
+                dataformats="HWC",
+            )
+            spec_predict = spec_predict[0].data.cpu().numpy()
+            tb_logger.add_image(
+                "val_mel_predicted", plot_spectrogram_to_numpy(spec_predict.T), self.global_step, dataformats="HWC",
+            )
+        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()  # This reduces across batches, not workers!
+        self.log('val_loss', avg_loss, sync_dist=True)
+
+        self.log_train_images = True
+
+    def parse(self, str_input: str) -> torch.tensor:
+        if str_input[-1] not in [".", "!", "?"]:
+            str_input = str_input + "."
+
+        # Convert text -> normalized text -> list of phones per word -> indices
+        norm_text = re.findall("""[\w']+|[.,!?;"]""", self.parser._normalize(str_input))
+        phones = [self.word2phones[t] for t in norm_text]
```

Comment:
Let's wrap this in a try, catch in case a word is not in cmudict, and leave some meaningful error message

Reply:
Good idea, updated.

Reply:
We should say how they can solve this error by adding a phone representation of the word to the dictionary

Reply:
Sounds good. Also added an optional argument for users to update the mapping.